Automattic/wp-calypso
# Invites: Fix post-invite-accept redirect

#### Changes proposed in this Pull Request

Currently, when accepting an invite we'll redirect to `/posts/:siteId` but without ensuring the user's sites have been refreshed. That leads the user to hang on the invitation page.

This PR changes the logic to wait a bit before the redirection, so re-fetching of sites will happen and the flow will work as expected.

We're also fixing an e2e test that was not expecting that the notice is displayed on the previous page.

Fixes #53546

#### Testing instructions

* Create an account X without sites. To do that, start at `/start`, register, and just stop the flow. Verify you're logged in by going to `wordpress.com`.
* From another account (Y) send an Editor invitation to X to join the site.
* Go to the inbox of user X and copy the invitation button link.
* Change the `wordpress.com` part of the link to the `calypso.live` link or `calypso.localhost` link (wherever you're testing this branch)
* Accept.
* Verify you're redirected to `/posts/:siteId` and the posts list loads.
